lib/cancan/model_adapters/active_record_5_adapter.rb

```diff
@@ -22,16 +22,13 @@ def self.matches_condition?(subject, name, value)
       private
 
       def build_joins_relation(relation, *where_conditions)
-        case CanCan.accessible_by_strategy
-        when :subquery
-          inner = @model_class.unscoped do
-            @model_class.left_joins(joins).where(*where_conditions)
-          end
-          @model_class.where(@model_class.primary_key => inner)
+        strategy_class.new(adapter: self, relation: relation, where_conditions: where_conditions).execute!
+      end
 
-        when :left_join
-          relation.left_joins(joins).distinct
-        end
+      def strategy_class
+        require_relative "strategies/#{CanCan.accessible_by_strategy}"
```

Comment:
Why not just require them at the top of the file?

Reply:
That would require code that the user in most cases would never use. I assume that most users are just going to set a single strategy and use that for everything :-)